components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityConfigParser.java

```diff
@@ -374,6 +386,60 @@ private void buildCookieConfig()    {
         }
     }
 
+    public void buildLegacyFeatureConfig() {
+
+        LegacyFeatureConfig legacyFeatureConfig;
+        OMElement legacyFeaturesConfigElement =
+                this.getConfigElement(IdentityConstants.LegacyFeatureConfigElements.LEGACY_FEATURE_CONFIG);
+        if (legacyFeaturesConfigElement != null) {
+
+            int legacyFeaturesConfigElementIndex = 0;
+            Iterator<OMElement> legacyFeatures = legacyFeaturesConfigElement.
+                    getChildrenWithName(getQNameWithIdentityNS(IdentityConstants.
+                            LegacyFeatureConfigElements.LEGACY_FEATURE));
+            if (legacyFeatures != null) {
+                while (legacyFeatures.hasNext()) {
+                    legacyFeaturesConfigElementIndex += 1;
+                    OMElement legacyFeature = legacyFeatures.next();
+
+                    OMElement legacyFeatureIdElement = legacyFeature.getFirstChildWithName(getQNameWithIdentityNS(
+                            IdentityConstants.LegacyFeatureConfigElements.LEGACY_FEATURE_ID));
+                    OMElement legacyFeatureVersionElement = legacyFeature.getFirstChildWithName(getQNameWithIdentityNS(
+                            IdentityConstants.LegacyFeatureConfigElements.LEGACY_FEATURE_VERSION));
+                    OMElement legacyFeatureEnableElement = legacyFeature.getFirstChildWithName(getQNameWithIdentityNS(
+                            IdentityConstants.LegacyFeatureConfigElements.LEGACY_FEATURE_ENABLE));
+
+                    String legacyFeatureId;
+                    String legacyFeatureVersion;
+                    boolean isLegacyFeatureEnable = false;
+                    if (legacyFeatureIdElement != null && legacyFeatureVersionElement != null
+                            && legacyFeatureEnableElement != null) {
+                        legacyFeatureId = legacyFeatureIdElement.getText();
+                        if (StringUtils.isNotBlank(legacyFeatureId)) {
+                            legacyFeatureId = legacyFeatureId.trim();
+                        }
+                        legacyFeatureVersion = legacyFeatureVersionElement.getText();
+                        if (StringUtils.isNotBlank(legacyFeatureVersion)) {
+                            legacyFeatureVersion = legacyFeatureVersion.trim();
+                        }
+                        String legacyFeatureEnable = legacyFeatureEnableElement.getText();
+                        if (StringUtils.isNotBlank(legacyFeatureEnable)) {
+                            isLegacyFeatureEnable = Boolean.parseBoolean(legacyFeatureEnable.trim());
+                        }
+
+                        String legacyFeatureConfigKey = legacyFeatureId + legacyFeatureVersion;
```

Comment:
wasn't better to add a seperater between the feature id and version.

Reply:
In https://github.com/GANGANI/carbon-identity-framework/blob/76ad8e0307b509aefc1c7e7716756ad143850105/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java#L228-L236, this Map value was used. Since some configs do not have a feature version, adding a separator would not be appropriate when retrieving the value from the Map. WDYT?